app/controllers/reactions_controller.rb

```diff
@@ -102,10 +102,11 @@ def create
     render json: { result: result, category: category }
   end
 
-  def cached_user_public_reactions(user)
-    Rails.cache.fetch("cached-user-reactions-#{user.id}-#{user.public_reactions_count}", expires_in: 24.hours) do
-      user.reactions.public_category
+  def cached_user_public_reactions(user, comment_ids)
+    cache = Rails.cache.fetch("cached-user-reactions-#{user.id}-#{user.public_reactions_count}", expires_in: 24.hours) do
+      JSON.parse(user.reactions.public_category.to_json)
```

Comment:
I would throw the `.where(reactable_id: comment_ids)` on the end of your AR relation here then `.to_json` it so you can skip the select below.

Reply:
Unfortunately `comment_ids` is the request's param and doesn't fall under `cached-user-reactions-#{user.id}-#{user.public_reactions_count` cache.

Reply:
Got it! Rather than doing `.to_json` here then `JSON.parse` you could do `user.reactions.public_category.map(&:attributes)` to return a hash. And actually, I am not sure how many comment_ids we can expect here but below looping through an array and comparing to another array is going to result in a lot of iterations. I think you could speed that up if you create a hash here instead of an array of hashes. So something like this
```ruby
user.reactions.public_category.each_with_object({}){|r, h| h[r.reactable_id] = r.attributes}

# below
cache.slice(*comment_ids)
```

Reply:
Thanks for this optimization tip! Updated.